### PR TITLE
Continue working on run-image.

### PR DESCRIPTION
**Pull request by floitsch**
Now assumes that the given dir uses the OTA structure.

### CODE REVIEW

File: tools/firmware.toit [hunk 1]
```diff
@@ -700,32 +700,32 @@ extract-host parsed/cli.Parsed envelope/Envelope --config-encoded/ByteArray:
 
   // Reserve space for the header.
   header-size := 6 * 4
-  bits.reserve header-size
+  bits.grow-by header-size
```
Comment (kasperl) [inline tools/firmware.toit:703]: expand?

File: tools/run-image.toit [hunk 2]
```diff
@@ -43,20 +57,179 @@ class SystemImage extends ContainerImage:
...
+
+class FirmwareWriter_ extends services.ServiceResource implements FirmwareWriter:
+  static image/ByteArray := #[]
+  view_/ByteArray? := null
+  cursor_/int := 0
+  ota-dir-inactive_/string
+
+  constructor provider/FirmwareServiceProvider client/int from/int to/int --ota-dir-inactive/string:
+    ota-dir-inactive_ = ota-dir-inactive
+    if to > image.size: image = image + (ByteArray to - image.size: random 0x100)
```
Comment (kasperl) [inline tools/run-image.toit:111]: Add a comment to explain why you do this random padding?

**Review by kasperl — APPROVED** [review]
Looks good to me!

File: tools/firmware.toit (hunk 1 above)
Comment (floitsch) [reply tools/firmware.toit:703]: I think "grow-by" is fine, and I would prefer not to change the `io.Writer`.

File: tools/run-image.toit (hunk 2 above)
Comment (floitsch) [reply tools/run-image.toit]: done.

**Pull request merged**